lib/checkbool.cpp

```diff
@@ -454,3 +455,27 @@ void CheckBool::assignBoolToFloatError(const Token *tok)
     reportError(tok, Severity::style, "assignBoolToFloat",
                 "Boolean value assigned to floating point variable.", CWE704, false);
 }
+
+void CheckBool::returnValueOfFunctionReturningBool(void)
+{
+    if (!mSettings->isEnabled(Settings::STYLE))
+        return;
+
+    const SymbolDatabase * const symbolDatabase = mTokenizer->getSymbolDatabase();
+
+    for (const Scope * scope : symbolDatabase->functionScopes) {
+        if (!(scope->function && Token::Match(scope->function->retDef, "bool|_Bool")))
+            continue;
+
+        for (const Token* tok = scope->bodyStart->next(); tok && (tok != scope->bodyEnd); tok = tok->next()) {
+            if (Token::simpleMatch(tok, "return") &&
+                (tok->astOperand1()->getValueGE(2, mSettings) || tok->astOperand1()->getValueLE(-1, mSettings)))
```

Comment:
This will crash if there is a return without operands.

    bool aa() { return; }

Reply:
Oops, fixed.